api/post.go

```diff
@@ -561,7 +561,7 @@ func getExplicitMentions(message string, keywords map[string][]string) (map[stri
 }
 
 func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *model.Channel) {
-	pchan := Srv.Store.User().GetProfilesInChannel(channel.Id)
+	pchan := Srv.Store.User().GetProfilesInChannel(channel.Id, -1, -1)
```

Comment:
this will need a rebase after the PR from @coreyhulen  is merged or the other way around, I believe it now checks if it has to invalidate the cache or something like that